manager/scheduler/scheduler_test.go

```diff
@@ -1689,6 +1689,159 @@ func TestSchedulerPreexistingDeadTask(t *testing.T) {
 	assert.Equal(t, "id1", assignment.NodeID)
 }
 
+func TestSchedulerCompatiblePlatform(t *testing.T) {
+	ctx := context.Background()
+	// create tasks
+	// task1 - has a node it can run on
+	task1 := &api.Task{
+		ID:           "id1",
+		DesiredState: api.TaskStateRunning,
+		ServiceAnnotations: api.Annotations{
+			Name: "name1",
+		},
+		Status: api.TaskStatus{
+			State: api.TaskStatePending,
+		},
+		Spec: api.TaskSpec{
+			Placement: &api.Placement{
+				Platforms: []*api.Platform{
+					{
+						Architecture: "amd64",
+						OS:           "linux",
+					},
+				},
+			},
+		},
+	}
+
+	// task2 - has no node it can run on
+	task2 := &api.Task{
+		ID:           "id2",
+		DesiredState: api.TaskStateRunning,
+		ServiceAnnotations: api.Annotations{
+			Name: "name2",
+		},
+		Status: api.TaskStatus{
+			State: api.TaskStatePending,
+		},
+		Spec: api.TaskSpec{
+			Placement: &api.Placement{
+				Platforms: []*api.Platform{
+					{
+						Architecture: "intel",
+						OS:           "linux",
+					},
+				},
+			},
+		},
+	}
+
+	// task3 - no platform constraints, should run on any node
```

Comment:
I think you should add another test with task specifying multiple platforms **explicitly**. And verify it can be scheduled to nodes with different architecture.